rules/alerting_test.go

```diff
@@ -142,3 +143,96 @@ func TestAlertingRuleLabelsUpdate(t *testing.T) {
 		testutil.Equals(t, result, filteredRes)
 	}
 }
+
+func TestAlertingRuleExternalLabelsInTemplate(t *testing.T) {
+	suite, err := promql.NewTest(t, `
+		load 1m
+			http_requests{job="app-server", instance="0"}	75 85 70 70
+	`)
+	testutil.Ok(t, err)
+	defer suite.Close()
+
+	err = suite.Run()
+	testutil.Ok(t, err)
+
+	expr, err := promql.ParseExpr(`http_requests < 100`)
+	testutil.Ok(t, err)
+
+	ruleWithoutExternalLabels := NewAlertingRule(
+		"ExternalLabelDoesNotExist",
+		expr,
+		time.Minute,
+		labels.FromStrings("external_label", "{{ $externalLabels.foo }}"),
+		nil,
+		nil,
+		true, log.NewNopLogger(),
+	)
+	ruleWithExternalLabels := NewAlertingRule(
+		"ExternalLabelExists",
+		expr,
+		time.Minute,
+		labels.FromStrings("external_label", "{{ $externalLabels.foo }}"),
+		nil,
+		labels.FromStrings("foo", "bar"),
+		true, log.NewNopLogger(),
+	)
+	result := promql.Vector{
+		{
+			Metric: labels.FromStrings(
+				"__name__", "ALERTS",
+				"alertname", "ExternalLabelDoesNotExist",
+				"alertstate", "pending",
+				"instance", "0",
+				"job", "app-server",
+				"external_label", "",
```

Comment:
Hmm, does this mean we're sending an empty label rather than no label?

Reply:
No, it means that accessing a non-existent key in a Go map results in the zero value of the value (in this case an empty string). I think my naming of the label is causing the misunderstanding here. I'll change it and add some more things to the template to test more and better demonstrate what we are testing.